src/main/java/com/cloudbees/jenkins/support/filter/DefaultContentFilter.java

```diff
@@ -0,0 +1,270 @@
+
+import com.cloudbees.jenkins.support.SupportPlugin;
+import hudson.Extension;
+import hudson.ExtensionList;
+import hudson.Functions;
+import hudson.XmlFile;
+import hudson.model.AbstractItem;
+import hudson.model.ManagementLink;
+import hudson.model.Saveable;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.kohsuke.randname.RandomNameGenerator;
+import org.kohsuke.stapler.DataBoundSetter;
+import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.interceptor.RequirePOST;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nonnull;
+import javax.servlet.ServletException;
+import java.io.File;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * Filters contents based on names provided by all {@linkplain SensitiveNameProvider known sources}.
+ *
+ * @author Matt Sicker
+ * @see SensitiveNameProvider
+ * @since 2.48
+ */
+@Extension
+public class DefaultContentFilter extends ManagementLink implements ContentFilter, Saveable {
+
+    public static DefaultContentFilter get() {
+        return all().get(DefaultContentFilter.class);
+    }
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultContentFilter.class);
+    private static final String ALT_SEPARATOR = " » ";
+
+    private volatile boolean enabled;
+    private final Map<String, Replacer> mappings = new ConcurrentHashMap<>();
+    private final RandomNameGenerator randomNameGenerator = new RandomNameGenerator();
+    private final Set<String> stopWords = ConcurrentHashMap.newKeySet();
+
+    public DefaultContentFilter() {
+        stopWords.addAll(Arrays.asList(
+                "jenkins", "node", "master", "computer",
+                "item", "label", "view", "all", "unknown",
+                "user", "anonymous", "authenticated"
+        ));
+        load();
+    }
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    @DataBoundSetter
+    public void setEnabled(boolean enabled) {
+        this.enabled = enabled;
+        save();
+    }
+
+    public Map<String, String> getMappings() {
+        Map<String, String> map = new HashMap<>(mappings.size());
+        for (Map.Entry<String, Replacer> entry : mappings.entrySet()) {
+            map.put(entry.getKey(), entry.getValue().replacement);
+        }
+        return map;
+    }
+
+    public Set<String> getStopWords() {
+        return stopWords;
+    }
+
+    @CheckForNull
+    @Override
+    public Permission getRequiredPermission() {
```

Comment:
TL;DR: `@Nullable` for a value that could be null under specific circumstances compared to `@CheckForNull` that allows the value to be null in any circonstance.

---

The meaning is different and always confusing, already had discussion about that with Daniel and Devin.

The conclusion was: `@CheckForNull` implies you need to check the value before using, and `@Nullable` means the `null` value is totally fine. In general with `@Nullable` you have either an implicit way to determine the case the value is null depending on the arguments of the method or the "rule" is documented. 
Imagine you have a method `UserUtil.getFirstName(User user)`, if you specify that the firstName is always non-null when the user is non-null, so you can put `@Nullable` on the method, and in your usage, you check for the user to be null, otherwise you know the firstName will be non-null.

I read also that the `@Nullable` was just created to override a method that was `@Nullable` in its parent / interface, but that seems against OOP.

PS: `@Nullable` is not taken into consideration by FindBugs compared to `@CheckForNull` as it cannot understand the rule.

Reply:
I think FindBugs requires null checks for things annotated with `@CheckForNull` but not for `@Nullable`, so `@CheckForNull` is preferred. I'm not 100% sure though.

Reply:
Oh interesting, I think I understand.